pkg/core/workflow_store.go

```diff
@@ -62,44 +62,85 @@ func (ws *WorkflowStore) AddWorkflow(ctx context.Context, w *workflow.Workflow)
 	return w, nil
 }
 
+// DeleteWorkflow deletes the workflow from the store
+func (ws *WorkflowStore) DeleteWorkflow(ctx context.Context, name string) error {
+	sw, found := ws.items[name]
+	if !found {
+		return fmt.Errorf("workflow not found")
```

Comment:
Should it be treated as error?

Reply:
I think so, if you try to delete something that does not exist it should return a `404`.

Reply:
We've already agreed in some other case (deleting k3s resource when deleting an application) that when the resource that is supposed to be deleted is not found, it does not need to raise an error

Reply:
Alright, can I find this agreement somewhere?

Reply:
https://github.com/fuseml/fuseml-core/pull/31#discussion_r629959926

_"deleting a resource that has already been deleted by other means (e.g. another similar REST API delete call or deleted manually) shouldn't be treated as a failure"_